File: src/main/java/ru/javarush/cryptoanalyzer/konovalov/view/ConsoleApp.java
package ru.javarush.cryptoanalyzer.konovalov.view;


import java.util.Arrays;

import ru.javarush.cryptoanalyzer.konovalov.controller.MainController;
import ru.javarush.cryptoanalyzer.konovalov.entity.*;

import static ru.javarush.cryptoanalyzer.konovalov.view.Messages.*;


public class ConsoleApp {
    private MainController mainController;
    private Menu menu;

    public ConsoleApp(MainController mainController, Menu menu) {
        this.mainController = mainController;
        this.menu = menu;
    }


    public void run(String[] args) {
        Result result;

        do {
            if (args.length == 0) {
                args = menu.getArgs();
            }
            result = getResult(args);
            print(result);
            args = new String[0];

        }
        while (result.RESULT_CODE == ResultCode.ERROR);
    }

    private Result getResult(String[] args) {
        String action = args[0];
        String[] parameters = Arrays.copyOfRange(args, 1, args.length);

        return mainController.doAction(action, parameters);
    }

    private void print(Result result) {
        String message = switch (result.RESULT_CODE) {
            case OK -> String.format(OK_FORMAT, result.MESSAGE);
            case ERROR -> String.format(ERROR_FORMAT, result.MESSAGE);
        };
        System.out.println(message);
    }
}

File: src/main/kotlin/view/ConsoleApp.kt
package view

import controller.MainController
import entity.Result
import entity.ResultCode
import java.util.*

@Suppress("NAME_SHADOWING")
class ConsoleApp(private val mainController: MainController, private val menu: Menu) {

fun run(args: Array<String>){
    var result: Result
    var args = args

    do{
        if(args.isEmpty()){
            args = menu.getArgs()
        }
        result =getResult(args)
        print(result)
    }
        while(result.RESULT_CODE == ResultCode.ERROR)

}

    private fun getResult(args: Array<String>): Result {
        val action : String = args[0]
        val parameters : Array<String> = args.copyOfRange(1, args.size)

        return action.let { mainController.doAction(it,parameters) }
    }

}